Classes/iOSDeviceManagerLogging.m

```diff
@@ -0,0 +1,230 @@
+                                          contents:nil
+                                        attributes:attributes];
+
+
+  [self deleteOldLogFiles];
+
+  NSString *currentSymlink = [logsDirectory stringByAppendingPathComponent:@"current.log"];
+  [[NSFileManager defaultManager] removeItemAtPath:currentSymlink error:nil];
+
+  [[NSFileManager defaultManager] createSymbolicLinkAtPath:currentSymlink
+                                       withDestinationPath:path error:nil];
+
+  return path;
+}
+
+- (BOOL)isLogFile:(NSString *)fileName {
+
+  BOOL hasProperPrefix = [fileName hasPrefix:IDMLogFilePrefix];
+  BOOL hasProperSuffix = [fileName hasSuffix:@".log"];
+  BOOL hasProperDate = NO;
+
+  if (hasProperPrefix && hasProperSuffix) {
+    NSUInteger lengthOfMiddle = fileName.length - IDMLogFilePrefix.length - @".log".length;
+
+    // Date string should have at least 19 characters: "2013-12-03-17-14-10"
+    if (lengthOfMiddle >= 19) {
+      NSRange range = NSMakeRange(IDMLogFilePrefix.length, lengthOfMiddle);
+
+      NSString *middle = [fileName substringWithRange:range];
+      NSArray *components = [middle componentsSeparatedByString:@"-"];
+
+      if (components.count == 6) {
+        NSDateFormatter *dateFormatter = [self logFileDateFormatter];
+        NSDate *date = [dateFormatter dateFromString:middle];
+
+        if (date) {
+          hasProperDate = YES;
+        }
+      }
+    }
+  }
+
+  return (hasProperPrefix && hasProperDate && hasProperSuffix);
+}
+
+
+@end
+
+static NSString *const IDMLogFormatterDateFormat = @"yyyy-MM-dd HH:mm:ss.SSS";
+static NSString *const IDMLogFormatterDateFormatterKey = @"sh.calaba-IDMLogFormatter-NSDateFormatter";
+
+@interface IDMLogFormatter : NSObject <DDLogFormatter>  {
+  int32_t atomicLoggerCount;
+  NSDateFormatter *threadUnsafeDateFormatter;
+}
+
+- (NSString *)stringFromDate:(NSDate *)date;
+
+@end
+
+
+@implementation IDMLogFormatter
+
+- (NSString *)stringFromDate:(NSDate *)date {
+  int32_t loggerCount = OSAtomicAdd32(0, &atomicLoggerCount);
+
+  if (loggerCount <= 1) {
+    // Single-threaded mode.
+
+    if (!threadUnsafeDateFormatter) {
+      threadUnsafeDateFormatter = [[NSDateFormatter alloc] init];
+      [threadUnsafeDateFormatter setDateFormat:IDMLogFormatterDateFormat];
+    }
+
+    return [threadUnsafeDateFormatter stringFromDate:date];
+  } else {
+    // Multi-threaded mode.
+    // NSDateFormatter is NOT thread-safe.
+    NSMutableDictionary *threadDictionary = [[NSThread currentThread] threadDictionary];
+    NSDateFormatter *dateFormatter = threadDictionary[IDMLogFormatterDateFormatterKey];
+
+    if (dateFormatter == nil) {
+      dateFormatter = [[NSDateFormatter alloc] init];
+      [dateFormatter setDateFormat:IDMLogFormatterDateFormat];
+
+      threadDictionary[IDMLogFormatterDateFormatterKey] = dateFormatter;
+    }
+
+    return [dateFormatter stringFromDate:date];
+  }
+}
+
+- (NSString *)formatLogMessage:(DDLogMessage *)logMessage {
```

Comment:
```
# Examples
2016-11-11 16:20:07.199 DEBUG FBControlCoreLogger:130 | 
2016-11-10 16:24:02.679 ERROR ConsoleWriter:24 |
```

It is a bit awkward because most of the logging in FB is done through FBControlCoreLogger:130 which does not give us much information.